src/test_driver/openiotsdk/integration-tests/common/utils.py

```diff
@@ -19,38 +19,83 @@
 import logging
 import random
 import re
+import traceback
+from typing import List
 
 from chip import discovery, exceptions
 from chip.clusters import Attribute as ClusterAttribute
 from chip.clusters import Objects as GeneratedObjects
 from chip.interaction_model import delegate as IM
 from chip.setup_payload import SetupPayload
 
+
+from pyedmgr import TestDevice
+
 log = logging.getLogger(__name__)
 
 IP_ADDRESS_BUFFER_LEN = 100
 
 
-def get_setup_payload(device):
+def strip_escape(string_to_escape) -> str:
+    raw_ansi_pattern = r'\033\[((?:\d|;)*)([a-zA-Z])'
+    ansi_pattern = raw_ansi_pattern.encode()
+    ansi_eng = re.compile(ansi_pattern)
+    matches = []
+    for match in ansi_eng.finditer(string_to_escape):
+        matches.append(match)
+    matches.reverse()
+    for match in matches:
+        start = match.start()
+        end = match.end()
+        string_to_escape = string_to_escape[0:start] + string_to_escape[end:]
```

Comment:
I feel like the indexes here might be wrong. As we proceed through the matches, the sting to escape become shorter and shorter. Yet the indexed of start and end would be the indexes in the original string ?

Reply:
wouldn't it be easier to just regex search and replace (re.sub) any illegal character into a ` `?

Reply:
Both valid points. I'll try replacing it with `re.sub`.